src/components/Table/Table.tsx

```diff
@@ -2,17 +2,27 @@ import type { HTMLProps } from 'react';
 import React from 'react';
 import cn from 'classnames';
 
+import { useMediaQuery } from '@/hooks/useMediaQuery';
+import { tokens } from '@/DesignTokens';
+
 import classes from './Table.module.css';
 import type { ChangeHandler } from './Context';
 import { TableContext } from './Context';
 import type { RowData } from './TableRow';
 
+export enum ScreenSize {
+  Mobile = 'mobile',
+  Tablet = 'tablet',
+  Laptop = 'laptop',
+}
+
 export interface TableProps
   extends Omit<HTMLProps<HTMLTableElement>, 'onChange'> {
   children?: React.ReactNode;
   selectRows?: boolean;
   onChange?: ChangeHandler;
   selectedValue?: RowData;
+  screenSize?: ScreenSize;
```

Comment:
Why do you have to pass in the screen size here as a property? I see you imported `useMediaQuery` and `tokens` above, but never used them, so I'm guessing you tried it out. If there are specific reasons the screen size needs to be an externally provided property (i.e. it is not purely controlled by the actual screen size), I think that should be explained somewhere.